### PR TITLE
migrate the /payment/[uid] page

### PR DESCRIPTION
**Pull request by grzpab**

### CODE REVIEW

File: apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+              successRedirectUrl: true,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  if (!rawPayment) {
+    return notFound();
```
Comment (DmytroHryshyn) [inline apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx:111]: Do we need to use `return` before notFound()?  I think we can just do 
```
 if (!rawPayment) {
    notFound();
    }

File: apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  }
+
+  return {
+    user,
+    eventType: {
+      ...eventType,
+      metadata: EventTypeMetaDataSchema.parse(eventType.metadata),
+    },
+    booking,
+    trpcState: ssr.dehydrate(),
```
Comment (DmytroHryshyn) [inline apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx:164]: it should be awaited `await ssr.dehydrate()`

File: apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx (hunk 1 above)
Comment (grzpab) [reply apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx:111]: does it throw an error and finish the execution? how does it work?

File: apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx (hunk 2 above)
Comment (grzpab) [reply apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx:164]: sure, will check if it works after the change.

File: packages/app-store/index.ts [hunk 3]
```diff
@@ -40,4 +40,13 @@ const appStore = {
   shimmervideo: () => import("./shimmervideo"),
 };
 
-export default appStore;
+const exportedAppStore: typeof appStore & {
+  ["mock-payment-app"]?: () => Promise<typeof import("./mock-payment-app/index")>;
+} = appStore;
+
+// used only in tests
+// if (process.env.MOCK_PAYMENT_APP_ENABLED) {
```
Comment (DmytroHryshyn) [inline packages/app-store/index.ts:48]: Is this commented on purpose?

Comment (grzpab) [reply packages/app-store/index.ts:48]: good point, applied a fix

File: apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx (hunk 1 above)
Comment (DmytroHryshyn) [reply apps/web/app/future/(individual-page-wrapper)/payment/[uid]/page.tsx:111]: yes, it throws error
```
function notFound() {
    // eslint-disable-next-line no-throw-literal
    const error = new Error(NOT_FOUND_ERROR_CODE);
    error.digest = NOT_FOUND_ERROR_CODE;
    throw error;
}
```

**Review by DmytroHryshyn — APPROVED** [review]

**Pull request merged**